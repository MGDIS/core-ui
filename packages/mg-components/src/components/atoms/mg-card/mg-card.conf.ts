/**
 * List available variants
 */
export const variants = ['info', 'warning', 'success', 'danger', 'app'] as const;

/**
 * Variant type
 */
export type VariantType = (typeof variants)[number];

/**
 * List available variant styles
 */
export const variantStyles = ['bar-left', 'fill'] as const;

/**
 * VariantStyle type
 */
export type VariantStyleType = (typeof variantStyles)[number];
