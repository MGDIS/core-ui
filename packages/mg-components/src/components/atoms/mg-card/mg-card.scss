@import '../../../styles/mixins/reset-mg-card-slotted-variables.scss';

@function getColorPrefix($color) {
  @if $color == 'app' {
    @return 'mg-color';
  }
  @return 'color';
}

@mixin card-variants($variants...) {
  @each $variant in $variants {
    &.mg-card--#{$variant} {
      overflow: hidden;
      &.mg-card--bar-left .mg-card__bar {
        background-color: hsl(var(--#{getColorPrefix($variant)}-#{$variant}-h), var(--#{getColorPrefix($variant)}-#{$variant}-s), var(--#{getColorPrefix($variant)}-#{$variant}-l));
      }
      &.mg-card--fill {
        background-color: hsl(var(--#{getColorPrefix($variant)}-#{$variant}-h), var(--#{getColorPrefix($variant)}-#{$variant}-s), var(--mg-color-soft-l));
      }
    }
  }
}

.mg-card {
  // Display
  position: relative;
  width: 100%;
  padding: var(--mg-card-padding, var(--mg-card-padding-default));
  overflow: var(--mg-card-overflow, var(--mg-card-overflow-default));
  box-sizing: border-box;
  z-index: inherit;
  // Decoration
  background: var(--mg-card-background, var(--mg-card-background-default));
  border: var(--mg-card-border, var(--mg-card-border-default));
  border-radius: var(--mg-card-border-radius, var(--mg-card-border-radius-default));
  box-shadow: var(--mg-card-box-shadow, var(--mg-card-box-shadow-default));

  @include reset-slot-card-variables();
  @include card-variants('info', 'warning', 'success', 'danger', 'app');

  &__bar {
    position: absolute;
    top: 0;
    left: 0;
    width: 0.3rem;
    height: 100%;
  }
}
