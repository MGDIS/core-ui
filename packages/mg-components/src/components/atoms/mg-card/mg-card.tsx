import { Component, h, Prop, State, Watch } from '@stencil/core';
import { ClassList } from '@mgdis/core-ui-helpers/dist/utils';

/**
 * @slot - Card content
 */
@Component({
  tag: 'mg-card',
  styleUrl: '../../../../node_modules/@mgdis/styles/dist/components/mg-card.css',
  shadow: true,
})
export class MgCard {
  /**
   * Add box-shadow style
   */
  @Prop() shadow = false;
  @Watch('shadow')
  watchShadow(newValue: boolean): void {
    if (newValue) {
      this.classCollection.add('mg-c-card--shadow');
    } else {
      this.classCollection.delete('mg-c-card--shadow');
    }
  }

  /**
   * Define the border radius size
   */
  @Prop() radius: 'small' | 'medium' | 'large' = 'large';
  @Watch('radius')
  watchRadius(newValue: 'small' | 'medium' | 'large'): void {
    // Remove all possible radius classes
    this.classCollection.delete('mg-c-card--radius-small');
    this.classCollection.delete('mg-c-card--radius-medium');
    // Add new class only if radius is valid and not large (default)
    if (newValue === 'small' || newValue === 'medium') {
      this.classCollection.add(`mg-c-card--radius-${newValue}`);
    }
  }

  /**
   * Component classes
   */
  @State() classCollection: ClassList = new ClassList(['mg-c-card']);

  /*************
   * Lifecycle *
   *************/

  /**
   * Check if component props are well configured on init
   */
  componentWillLoad(): void {
    this.watchShadow(this.shadow);
    this.watchRadius(this.radius);
  }

  /**
   * Render
   * @returns HTML Element
   */
  render(): HTMLElement {
    return (
      <div class={this.classCollection.join()}>
        <slot></slot>
      </div>
    );
  }
}
