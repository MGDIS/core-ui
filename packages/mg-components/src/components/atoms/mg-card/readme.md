## Specifications

### Style

![](./mg-card/doc/img/mg-card-style.png)

1. Border radius: 5px
2. Background: [@color-light](./?path=/docs/style-colors--page)
3. Border: 1px, [@color-dark](./?path=/docs/style-colors--page) with alpha at 5%
4. Shadow: [@shadow](./?path=/docs/style-colors--page)

### Variant on bar / on background

You can set variant color on left bar or on background of the card

Values for variant colors are : 
- info
- success
- warning
- danger
- app

![](./mg-card/doc/img/mg-card-variant.png)

*exemple of "app color" for the last one*

Variant applied on the left bar uses **[full colors](./?path=/docs/style-colors--page)**

Variant applied on the background uses **[soft colors](./?path=/docs/style-colors--page)**

### Spacing

![](./mg-card/doc/img/mg-card-spacing.png)

A padding of 16px is applied around the content

### Size

![](./mg-card/doc/img/mg-card-size.png)

1. Ajusting with the content
2. Ajusting with its parent (100%)

## CSS Variables

If needed some [variables](./?path=/story/css-variables--page) are available to customize the component:

- `--mg-card-padding`: define car padding, default: `1.6rem`
- `--mg-card-border-radius`: define card border radius, default: `0.5rem`
- `--mg-card-background`: define card background, default: `hsl(var(--color-light))`
- `--mg-card-border`: define card border, default: `0.1rem solid hsla(var(--color-dark), 5%)`
- `--mg-card-box-shadow`: define card shadow, default: `var(--box-shadow)`
- `--mg-card-box-overflow`: define card overflow, default: `unset`

<!-- Auto Generated Below -->


## Properties

| Property       | Attribute       | Description                                 | Type                                                    | Default     |
| -------------- | --------------- | ------------------------------------------- | ------------------------------------------------------- | ----------- |
| `variant`      | `variant`       | Define variant prop Default: undefined      | `"app" \| "danger" \| "info" \| "success" \| "warning"` | `undefined` |
| `variantStyle` | `variant-style` | Define variantStyle prop Default: undefined | `"bar-left" \| "fill"`                                  | `undefined` |


## Dependencies

### Used by

 - [mg-message](../../molecules/mg-message)
 - [mg-modal](../../molecules/mg-modal)
 - [mg-panel](../../molecules/mg-panel)
 - [mg-popover](../../molecules/mg-popover)

### Graph
```mermaid
graph TD;
  mg-message --> mg-card
  mg-modal --> mg-card
  mg-panel --> mg-card
  mg-popover --> mg-card
  style mg-card fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
