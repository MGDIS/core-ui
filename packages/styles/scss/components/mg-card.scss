@use 'mixins/reset-mg-card-slotted-variables';

/*!
 * @prop --mg-c-card-overflow: Defines the overflow property for the card. Unset by default, fallback on `--mg-c-card-overflow-default`.
 * @prop --mg-c-card-overflow-default: Defines the default overflow property for the card. Unset by default.
 * @prop --mg-c-card-min-width: Defines the min-width property for the card. Unset by default.
 * @prop --mg-c-card-max-width: Defines the max-width property for the card. Unset by default.
 * @prop --mg-c-card-spacing: Defines the padding property for the card. Unset by default, fallback on `--mg-c-card-spacing-default`.
 * @prop --mg-c-card-spacing-default: Defines the default padding property for the card. Default value: `--mg-b-size-16`.
 * @prop --mg-c-card-border: Defines the border property for the card. Unset by default, fallback on `--mg-c-card-border-default`.
 * @prop --mg-c-card-border-default: Defines the default border property for the card. Default value: `var(--mg-b-size-border) solid var(--mg-b-color-neutral-30)`.
 * @prop --mg-c-card-border-radius: Defines the border radius property for the card. Unset by default, fallback on `--mg-c-card-border-radius-default`.
 * @prop --mg-c-card-border-radius-default: Defines the default border radius property for the card. Default value: `--mg-b-size-12`.
 * @prop --mg-c-card-color-background: Defines the background property for the card. Unset by default, fallback on `--mg-c-card-color-background-default`.
 * @prop --mg-c-card-color-background-default: Defines the default background property for the card. Default value: `--mg-b-color-light`.
 * @prop --mg-c-card-box-shadow: Defines the box shadow property for the card. Unset by default, fallback on `--mg-c-card-box-shadow-default`.
 * @prop --mg-c-card-box-shadow-default: Defines the default box shadow property for the card. Default value: `--mg-b-box-shadow`.
 */

.mg-c-card {
  position: relative;
  z-index: inherit;
  overflow: var(--mg-c-card-overflow, var(--mg-c-card-overflow-default));
  width: 100%;
  min-width: var(--mg-c-card-min-width);
  max-width: var(--mg-c-card-max-width);
  box-sizing: border-box;
  padding: var(--mg-c-card-spacing, var(--mg-c-card-spacing-default));
  border: var(--mg-c-card-border, var(--mg-c-card-border-default));
  border-radius: var(--mg-c-card-border-radius, var(--mg-c-card-border-radius-default));
  background: var(--mg-c-card-color-background, var(--mg-c-card-color-background-default));

  @include reset-mg-card-slotted-variables.reset-slot-card-variables;

  &.mg-c-card--shadow {
    box-shadow: var(--mg-c-card-box-shadow, var(--mg-c-card-box-shadow-default));
  }

  &.mg-c-card--radius-xsmall {
    --mg-c-card-border-radius: var(--mg-b-size-4);
  }

  &.mg-c-card--radius-small {
    --mg-c-card-border-radius: var(--mg-b-size-8);
  }
}
