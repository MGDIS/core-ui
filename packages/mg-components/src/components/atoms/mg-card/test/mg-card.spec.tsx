import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgCard } from '../mg-card';
import { variantStyles, variants } from '../mg-card.conf';

const getPage = (props, slot) =>
  newSpecPage({
    components: [MgCard],
    template: () => <mg-card {...props}>{slot}</mg-card>,
  });

describe('mg-card', () => {
  test.each([undefined, ...variants.flatMap(variant => variantStyles.map(variantStyle => ({ variant, variantStyle })))])('render', async props => {
    const { root } = await getPage(
      props,
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    );
    expect(root).toMatchSnapshot();
  });

  test.each([
    { oldValue: { variant: variants[0], variantStyle: variantStyles[0] }, newValue: { variant: variants[1], variantStyle: variantStyles[0] } },
    { oldValue: { variant: variants[0], variantStyle: variantStyles[0] }, newValue: { variant: variants[0], variantStyle: variantStyles[1] } },
  ])('should update classList', async ({ oldValue, newValue }) => {
    const page = await getPage(
      oldValue,
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    );
    expect(page.root).toMatchSnapshot();

    const mgCard = page.doc.querySelector('mg-card');
    Object.keys(newValue).forEach(key => {
      mgCard[key] = newValue[key];
    });

    await page.waitForChanges();

    expect(page.root).toMatchSnapshot();
  });

  describe('errors', () => {
    test.each([
      { props: { variantStyle: 'batman' }, error: `<mg-card> prop "variantStyle" must match VariantStyleType type.` },
      ...variantStyles.map(variantStyle => ({ props: { variantStyle }, error: `<mg-card> prop "variantStyle" must be paired with ${JSON.stringify(variants)} "variant" prop.` })),
      { props: { variant: 'batman' }, error: `<mg-card> prop "variant" must match VariantType type.` },
      ...variants.map(variant => ({ props: { variant }, error: `<mg-card> prop "variant" must be paired with ${JSON.stringify(variantStyles)} "variantStyle" prop.` })),
    ])('should throw error %s', async ({ props, error }) => {
      expect.assertions(1);
      try {
        await getPage(
          props,
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
        );
      } catch (err) {
        expect(err.message).toMatch(error);
      }
    });
  });
});
