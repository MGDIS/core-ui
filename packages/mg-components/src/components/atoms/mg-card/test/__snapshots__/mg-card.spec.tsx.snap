// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-card render 1`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 2`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--info">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 3`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--fill mg-card--info">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 4`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--warning">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 5`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--fill mg-card--warning">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 6`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--success">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 7`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--fill mg-card--success">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 8`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--danger">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 9`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--danger mg-card--fill">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 10`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--app mg-card--bar-left">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 11`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--app mg-card--fill">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 12`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--info">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 13`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--warning">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 14`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--success">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 15`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--danger">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card render 16`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--app mg-card--bar-left">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card should update classCollection 1`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--info">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card should update classCollection 2`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--warning">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card should update classCollection 3`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--bar-left mg-card--info">
      <span class="mg-card__bar"></span>
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;

exports[`mg-card should update classCollection 4`] = `
<mg-card>
  <mock:shadow-root>
    <div class="mg-card mg-card--fill mg-card--info">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</mg-card>
`;
